src/moin/converter/_tests/test_markdown_in_out.py

```diff
@@ -0,0 +1,146 @@
+# Copyright: 2008 MoinMoin:BastianBlank
+# Copyright: 2010 MoinMoin:DmitryAndreev
+# Copyright: 2018 MoinMoin:RogerHaase - modified test_moinwiki_in_out.py for markdown
+# License: GNU GPL v2 (or any later version), see LICENSE.txt for details.
+
+"""
+MoinMoin - Tests for markdown->DOM->markdown using markdown_in and markdown_out converters
+
+TODO: Failing tests are commented out and need to be fixed.
+"""
+
+
+import pytest
+import re
+
+from emeraldtree import ElementTree as ET
+
+from moin.util.tree import moin_page, xlink, xinclude, html
+from moin.converter.markdown_in import Converter as conv_in
+from moin.converter.markdown_out import Converter as conv_out
+
+
+class TestConverter(object):
+
+    input_namespaces = 'xmlns="{0}" xmlns:page="{1}" xmlns:xlink="{2}" xmlns:xinclude="{3}" xmlns:html="{4}"'.format(
+        moin_page.namespace, moin_page.namespace, xlink.namespace, xinclude.namespace, html.namespace)
+
+    namespaces = {
+        moin_page.namespace: 'page',
+        xlink.namespace: 'xlink',
+        xinclude.namespace: 'xinclude',
+        html.namespace: 'html',
+    }
+    input_re = re.compile(r'^(<[a-z:]+)')
+    output_re = re.compile(r'\s+xmlns(:\S+)?="[^"]+"')
+
+    def setup_class(self):
+        self.conv_in = conv_in()
+        self.conv_out = conv_out()
+
+    def test_base(self):
+        data = [
+            (u'Text', 'Text\n'),
+            (u"Text\n\nText\n", 'Text\n\nText\n'),
+            (u"xxx\n\n------\n\n------\n\n------\n", 'xxx\n\n----\n\n----\n\n----\n'),
+            (u"----\n\n------\n\n--------\n", '----\n\n----\n\n----\n'),
+            (u"**strong**\n", "**strong**\n"),
+            (u"*emphasis*\n", "*emphasis*\n"),
+            (u"    blockcode\n", "    blockcode\n"),
+            (u"`monospace`\n", '`monospace`\n'),
+            (u"<strike>stroke</strike>\n", '<strike>stroke</strike>\n'),
+            # <ins> is changed to <u>
+            (u"<ins>underline</ins>\n", '<u>underline</u>\n'),
+            (u"<big>larger</big>\n", '<big>larger</big>\n'),
+            (u"<small>smaller</small>\n", '<small>smaller</small>\n'),
+            (u"<sup>super</sup>script\n", '<sup>super</sup>script\n'),
+            (u"<sub>sub</sub>script\n", '<sub>sub</sub>script\n'),
+        ]
+        for i in data:
+            yield (self.do, ) + i
```

Comment:
hmm, did you copy an old version of these tests? iirc, that `yield`-kind of tests is something modern pytest complains about because it is deprecated and I fixed all(?) of these recently.

Reply:
yes, started in June on hg repo